controllers/datadogagent/feature/dogstatsd/feature.go

```diff
@@ -0,0 +1,198 @@
+		f.udsHostFilepath = *config.Dogstatsd.UnixDomainSocket.HostFilepath
+	}
+	if apiutils.BoolValue(config.Dogstatsd.DogstatsdOriginDetection) {
+		f.originDetectionEnabled = true
+	}
+	if config.Dogstatsd.MapperProfiles != nil {
+		f.mapperProfiles = v1alpha1.ConvertCustomConfig(config.Dogstatsd.MapperProfiles)
+	}
+	reqComp = feature.RequiredComponents{
+		Agent: feature.RequiredComponent{
+			Containers: []apicommonv1.AgentContainerName{
+				apicommonv1.CoreAgentContainerName,
+			},
+		},
+	}
+	return reqComp
+}
+
+// ManageDependencies allows a feature to manage its dependencies.
+// Feature's dependencies should be added in the store.
+func (f *dogStatsDFeature) ManageDependencies(managers feature.ResourceManagers) error {
+	return nil
+}
+
+// ManageClusterAgent allows a feature to configure the ClusterAgent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *dogStatsDFeature) ManageClusterAgent(managers feature.PodTemplateManagers) error {
+	return nil
+}
+
+// ManageNodeAgent allows a feature to configure the Node Agent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *dogStatsDFeature) ManageNodeAgent(managers feature.PodTemplateManagers) error {
+	// udp
+	if f.hostPortEnabled {
+		if f.hostPortHostPort != 0 {
+			managers.Port().AddPortToContainer(apicommonv1.CoreAgentContainerName, &corev1.ContainerPort{
+				Name:          apicommon.DogStatsDHostPortName,
+				HostPort:      f.hostPortHostPort,
+				ContainerPort: apicommon.DogStatsDHostPortHostPort,
+				Protocol:      corev1.ProtocolUDP,
+			})
+		} else {
+			managers.Port().AddPortToContainer(apicommonv1.CoreAgentContainerName, &corev1.ContainerPort{
+				Name:          apicommon.DogStatsDHostPortName,
+				HostPort:      apicommon.DogStatsDHostPortHostPort,
+				ContainerPort: apicommon.DogStatsDHostPortHostPort,
+				Protocol:      corev1.ProtocolUDP,
+			})
+		}
+		managers.EnvVar().AddEnvVarToContainer(apicommonv1.CoreAgentContainerName, &corev1.EnvVar{
+			Name:  apicommon.DDDogStatsDNonLocalTraffic,
+			Value: "true",
+		})
+		// only add if uds origin detection is not enabled to prevent duplicates
+		if f.originDetectionEnabled && !f.udsEnabled {
+			managers.EnvVar().AddEnvVarToContainer(apicommonv1.CoreAgentContainerName, &corev1.EnvVar{
+				Name:  apicommon.DDDogstatsdOriginDetection,
+				Value: "true",
+			})
+		}
+	}
+
+	// uds
+	if f.udsEnabled {
+		socketVol, socketVolMount := volume.GetVolumes(apicommon.DogStatsDUDSSocketName, f.udsHostFilepath, f.udsHostFilepath, apicommon.DogStatsDUDSHostFilepathReadOnly)
+		managers.Volume().AddVolumeToContainer(&socketVol, &socketVolMount, apicommonv1.CoreAgentContainerName)
+		managers.EnvVar().AddEnvVarToContainer(apicommonv1.CoreAgentContainerName, &corev1.EnvVar{
+			Name:  apicommon.DDDogStatsDSocket,
+			Value: f.udsHostFilepath,
+		})
+		if f.originDetectionEnabled {
```

Comment:
to reduce confusion could we remove the `if` block on line 141 and have a separate top-level `if f. originDetectionEnabled` block that adds this envvar?

(i think you could also keep the `hostPID` setting in the same block, from looking at the helm chart it doesn't seem we distinguish between UDS or UDP origin detection when setting this)

Reply:
I thought `hostPID` was only used for [UDS origin detection](https://docs.datadoghq.com/developers/dogstatsd/unix_socket/?tab=kubernetes#origin-detection) since UDP relies on an `dd.internal.entity_id` tag instead. Would it still be safe to set `hostPID: true` for UDP as well?